src/client/power-pages/actions-hub/ActionsHubTreeDataProvider.ts

```diff
@@ -195,13 +291,35 @@ export class ActionsHubTreeDataProvider implements vscode.TreeDataProvider<Actio
                     currentEnvironmentName: authInfo!.OrganizationFriendlyName //Already checked in checkAuthInfo
                 };
 
-                if(!this._otherSites.length){
-                    return [new EnvironmentGroupTreeItem(currentEnvInfo, this._context, this._activeSites, this._inactiveSites)];
+                const rootItems: ActionsHubTreeItem[] = [];
+
+                rootItems.push(new EnvironmentGroupTreeItem(currentEnvInfo, this._context, this._activeSites, this._inactiveSites));
+                if(this._otherSites.length){
+                    rootItems.push(new OtherSitesGroupTreeItem(this._otherSites));
+                }
+
+                // Inject Metadata Diff root (if enabled & provider exists)
+                const mdProvider = ActionsHubTreeDataProvider._metadataDiffProvider;
+                if (mdProvider) {
+                    // Ensure provider builds diff list before evaluating hasData
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                    mdProvider.getChildren();
```

Comment:
Do we need to call getChildren manually? This is automatically called by the tree item when traversing the tree.